src/cobald/daemon/config/mapping.py

```diff
@@ -109,36 +111,79 @@ def load_name(absolute_name: str):
 
 
 class SectionPlugin(Generic[M]):
-    __slots__ = "section", "digest", "required"
+    """
+    Plugin to digest a top-level configuration section
+
+    :param section: Name of the section to digest
+    :param digest: callable that receive the section
+    :param required: whether the section must be present
+    """
 
-    __entry_point_flags__ = {"required"}
+    __slots__ = "section", "digest", "requirements"
+
+    @property
+    def required(self):
+        return self.requirements.required
+
+    @property
+    def before(self):
+        return self.requirements.before
+
+    @property
+    def after(self):
+        return self.requirements.after
 
     def __init__(
-        self, section: str, digest: Callable[[M], Any], required: bool = False
+        self, section: str, digest: Callable[[M], Any], requirements: PluginRequirements
     ):
         self.section = section
         self.digest = digest
-        self.required = required
+        self.requirements = requirements
```

Comment:
Please consider adding an assertion that `requirements` are not `None`.

Reply:
The way this class is constructed, ``None`` isn't an accepted parameter and won't be passed in unless someone messes with the rest of the code.